### PR TITLE
Check completeness of logs

### PR DESCRIPTION
**Pull request by HerbertJordan**
This PR adds an integration test verifying that log messages can be retrieved from the chain even after node restarts and re-initialization.

### CODE REVIEW

File: tests/block_header_test.go [hunk 1]
```diff
@@ -425,3 +439,93 @@ func getWitnessProofSteps(client *ethclient.Client, blockNumber int) ([]common.H
...
+	require := require.New(t)
+
+	numLogs := 0
+	for _, header := range headers {
+		receipts, err := client.BlockReceipts(context.Background(),
+			rpc.BlockNumberOrHashWithHash(header.Hash(), false))
+		require.NoError(err, "failed to get block receipts")
+
+		blockHash := header.Hash()
+		blockNumber := header.Number.Uint64()
```
Comment (facuMH) [inline tests/block_header_test.go:453]: why not do these declarations before getting the receipt and using the hash variable in `rpc.BlockNumberOrHashWithHash` ?

File: tests/block_header_test.go [hunk 2]
```diff
@@ -425,3 +439,93 @@ func getWitnessProofSteps(client *ethclient.Client, blockNumber int) ([]common.H
...
+func testHeaders_CanRetrieveLogEvents(t *testing.T, headers []*types.Header, client *ethclient.Client) {
+	require := require.New(t)
+
+	allLogs := []types.Log{}
+	for _, header := range headers {
+		receipts, err := client.BlockReceipts(context.Background(),
+			rpc.BlockNumberOrHashWithHash(header.Hash(), false))
+		require.NoError(err, "failed to get block receipts")
+
+		blockHash := header.Hash()
```
Comment (facuMH) [inline tests/block_header_test.go:482]: same as before, declare before fetching receipt, and use it for that as well

**Review by facuMH — APPROVED** [review]
Looks good, just a couple of minor suggestions

**Review by facuMH — APPROVED** [review]

**Review by jenikd — APPROVED** [review]
Looks good, thanks

**Pull request merged**